File: app/src/main/java/com/example/volumeintimemanager/FilesHandler.java
package com.example.volumeintimemanager;

import android.content.Context;
import android.util.Log;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStreamReader;

public class FilesHandler {
    Context _context;

    public FilesHandler(Context context){
        _context = context;
    }

    public String readFile(String fileName){
        try {
            FileInputStream fs = _context.openFileInput(fileName);
            InputStreamReader isr = new InputStreamReader(fs);
            BufferedReader bufferedReader = new BufferedReader(isr);
            StringBuilder sb = new StringBuilder();
            String line;

            while ((line = bufferedReader.readLine()) !=null){
                sb.append(line);
            }

            return sb.toString();
        } catch (FileNotFoundException fileNotFound){
            return null;
        } catch (IOException ioException){
            return null;
        }
    }

    public boolean createFile(String fileName, String jsonString){
        try{
            FileOutputStream fos = _context.openFileOutput(fileName, Context.MODE_PRIVATE);
            if(jsonString != null){
                fos.write(jsonString.getBytes());
            }
            fos.close();
            return true;
        } catch (FileNotFoundException fileNotFound){
            return false;
        } catch (IOException ioException) {
            return false;
        }
    }

    public boolean isFileAvailable(String fileName){
        String path = _context.getFilesDir().getAbsolutePath() + "/" + fileName;
        Log.d("FilesHandler", path);
        File file = new File(path);
        return file.exists();
    }
}

File: app/src/main/java/com/example/volumeintimemanager/FilesHandler.kt
package com.example.volumeintimemanager

import android.content.Context
import android.util.Log
import java.io.*

open class FilesHandler(@JvmField var _context: Context) {
    fun readFile(fileName: String?): String? {
        return try {
            val fs = _context.openFileInput(fileName)
            val isr = InputStreamReader(fs)
            val bufferedReader = BufferedReader(isr)
            val sb = StringBuilder()
            var line: String?
            while (bufferedReader.readLine().also { line = it } != null) {
                sb.append(line)
            }
            sb.toString()
        } catch (fileNotFound: FileNotFoundException) {
            null
        } catch (ioException: IOException) {
            null
        }
    }

    fun createFile(fileName: String?, jsonString: String?): Boolean {
        return try {
            val fos = _context.openFileOutput(fileName, Context.MODE_PRIVATE)
            if (jsonString != null) {
                fos.write(jsonString.toByteArray())
            }
            fos.close()
            true
        } catch (fileNotFound: FileNotFoundException) {
            false
        } catch (ioException: IOException) {
            false
        }
    }

    fun isFileAvailable(fileName: String): Boolean {
        val path = _context.filesDir.absolutePath + "/" + fileName
        Log.d("FilesHandler", path)
        val file = File(path)
        return file.exists()
    }
}